<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2024-02-06">

<title>Prepare data sets</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="prepare-data-sets_files/libs/clipboard/clipboard.min.js"></script>
<script src="prepare-data-sets_files/libs/quarto-html/quarto.js"></script>
<script src="prepare-data-sets_files/libs/quarto-html/popper.min.js"></script>
<script src="prepare-data-sets_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="prepare-data-sets_files/libs/quarto-html/anchor.min.js"></script>
<link href="prepare-data-sets_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="prepare-data-sets_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="prepare-data-sets_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="prepare-data-sets_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="prepare-data-sets_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">

<script src="prepare-data-sets_files/libs/kePrint-0.0.1/kePrint.js"></script>
<link href="prepare-data-sets_files/libs/lightable-0.0.1/lightable.css" rel="stylesheet">
<script src="prepare-data-sets_files/libs/htmlwidgets-1.6.3/htmlwidgets.js"></script>
<link href="prepare-data-sets_files/libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="prepare-data-sets_files/libs/datatables-binding-0.30/datatables.js"></script>
<script src="prepare-data-sets_files/libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<link href="prepare-data-sets_files/libs/dt-core-1.13.4/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="prepare-data-sets_files/libs/dt-core-1.13.4/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="prepare-data-sets_files/libs/dt-core-1.13.4/js/jquery.dataTables.min.js"></script>
<link href="prepare-data-sets_files/libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet">
<script src="prepare-data-sets_files/libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>


</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
  <nav id="TOC" role="doc-toc" class="toc-active" data-toc-expanded="3">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#clinical-data" id="toc-clinical-data" class="nav-link active" data-scroll-target="#clinical-data">Clinical data</a></li>
  <li><a href="#count-data" id="toc-count-data" class="nav-link" data-scroll-target="#count-data">Count data</a></li>
  <li><a href="#subset-count-data" id="toc-subset-count-data" class="nav-link" data-scroll-target="#subset-count-data">Subset count data</a></li>
  <li><a href="#annotation-data" id="toc-annotation-data" class="nav-link" data-scroll-target="#annotation-data">Annotation data</a></li>
  <li><a href="#save-data" id="toc-save-data" class="nav-link" data-scroll-target="#save-data">Save data</a></li>
  </ul>
</nav>
</div>
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Prepare data sets</h1>
</div>



<div class="quarto-title-meta">

    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">February 6, 2024</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<div class="cell" data-hash="prepare-data-sets_cache/html/setup-env_5d2360cd394b0425b4e3ae9f8023b9cb">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Set-up R environment</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list=</span><span class="fu">ls</span>())</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DT)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(biomaRt)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gtsummary)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(kableExtra)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="clinical-data" class="level1">
<h1>Clinical data</h1>
<div class="cell" data-hash="prepare-data-sets_cache/html/load-data_9f37234207158af798acefb12d34e94f">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># load meta data</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>file <span class="ot">&lt;-</span> <span class="st">"../data/box/metadata_ updated FB 20230504_to OD.xlsx"</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">read_xlsx</span>(file)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="co"># select columns</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>cols_v2_v3 <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"PatIDPx"</span>, <span class="st">"PatID"</span>, <span class="st">"age"</span>, </span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>             <span class="st">"BV_Diagnosis_v2...14"</span>, <span class="st">"Nugent_Score_v2"</span>,</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>             <span class="st">"BV_Diagnosis_v3...16"</span>, <span class="st">"Nugent_Score_v3"</span>,</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>             <span class="st">"Conc_Estradiol_v2 pg/ml"</span>, <span class="st">"Below_LLD_E2_V2"</span>,</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>             <span class="st">"Conc_Progesteron_v2 ng/ml"</span>,   <span class="st">"Below_LLD_P4_V2"</span>, </span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>             <span class="st">"Plasma_S_Estradiol_pg_mL_v3"</span>, <span class="st">"Below_LLD_E2_V3"</span>,</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>             <span class="st">"Plasma_S_Prog_ng_mL_v3"</span>,   <span class="st">"Below_LLD_P4_V3"</span></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>             )</span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a><span class="co">#Note: the lower limits of detection for E2 and P4 were 20 pg/ml and 0.09 ng/ml respectively. Values below this were reported as “below Lower Limit of Detection” and were assigned a value of ½ LLD, being 10 pg/ml and 0.05 ng/ml for E2 and P4 respectively.</span></span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a><span class="co"># format data in wide format</span></span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>df_clin <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span></span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">any_of</span>(cols_v2_v3)) <span class="sc">%&gt;%</span></span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">BV_v2 =</span> <span class="st">`</span><span class="at">BV_Diagnosis_v2...14</span><span class="st">`</span>, </span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a>         <span class="at">Nugent_v2 =</span> <span class="st">`</span><span class="at">Nugent_Score_v2</span><span class="st">`</span>, </span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a>         <span class="at">BV_v3 =</span> <span class="st">`</span><span class="at">BV_Diagnosis_v3...16</span><span class="st">`</span>, </span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a>         <span class="at">Nugent_v3 =</span> <span class="st">`</span><span class="at">Nugent_Score_v3</span><span class="st">`</span>,</span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a>         <span class="at">E2_v2 =</span> <span class="st">`</span><span class="at">Conc_Estradiol_v2 pg/ml</span><span class="st">`</span>, </span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a>         <span class="at">P4_v2 =</span> <span class="st">`</span><span class="at">Conc_Progesteron_v2 ng/ml</span><span class="st">`</span>, </span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a>         <span class="at">below_LLD_E2_v2 =</span> <span class="st">`</span><span class="at">Below_LLD_E2_V2</span><span class="st">`</span>, </span>
<span id="cb2-27"><a href="#cb2-27" aria-hidden="true" tabindex="-1"></a>         <span class="at">below_LLD_P4_v2 =</span> Below_LLD_P4_V2, </span>
<span id="cb2-28"><a href="#cb2-28" aria-hidden="true" tabindex="-1"></a>         <span class="at">E2_v3=</span> <span class="st">`</span><span class="at">Plasma_S_Estradiol_pg_mL_v3</span><span class="st">`</span>, </span>
<span id="cb2-29"><a href="#cb2-29" aria-hidden="true" tabindex="-1"></a>         <span class="at">P4_v3 =</span> <span class="st">`</span><span class="at">Plasma_S_Prog_ng_mL_v3</span><span class="st">`</span>, </span>
<span id="cb2-30"><a href="#cb2-30" aria-hidden="true" tabindex="-1"></a>         <span class="at">below_LLD_E2_v3 =</span> <span class="st">`</span><span class="at">Below_LLD_E2_V3</span><span class="st">`</span>, </span>
<span id="cb2-31"><a href="#cb2-31" aria-hidden="true" tabindex="-1"></a>         <span class="at">below_LLD_P4_v3 =</span> Below_LLD_P4_V3</span>
<span id="cb2-32"><a href="#cb2-32" aria-hidden="true" tabindex="-1"></a>         ) <span class="sc">%&gt;%</span></span>
<span id="cb2-33"><a href="#cb2-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">PatID =</span> <span class="fu">as.character</span>(PatID),</span>
<span id="cb2-34"><a href="#cb2-34" aria-hidden="true" tabindex="-1"></a>         <span class="at">age =</span> <span class="fu">as.numeric</span>(age), </span>
<span id="cb2-35"><a href="#cb2-35" aria-hidden="true" tabindex="-1"></a>         <span class="at">BV_v2 =</span> <span class="fu">as.factor</span>(<span class="fu">as.numeric</span>(BV_v2)), </span>
<span id="cb2-36"><a href="#cb2-36" aria-hidden="true" tabindex="-1"></a>         <span class="at">Nugent_v2 =</span> <span class="fu">as.numeric</span>(Nugent_v2), </span>
<span id="cb2-37"><a href="#cb2-37" aria-hidden="true" tabindex="-1"></a>         <span class="at">BV_v3 =</span> <span class="fu">as.factor</span>(<span class="fu">as.numeric</span>(BV_v3)), </span>
<span id="cb2-38"><a href="#cb2-38" aria-hidden="true" tabindex="-1"></a>         <span class="at">Nugent_v3 =</span> <span class="fu">as.numeric</span>(Nugent_v3), </span>
<span id="cb2-39"><a href="#cb2-39" aria-hidden="true" tabindex="-1"></a>         <span class="at">E2_v2 =</span> <span class="fu">as.numeric</span>(E2_v2), </span>
<span id="cb2-40"><a href="#cb2-40" aria-hidden="true" tabindex="-1"></a>         <span class="at">P4_v2 =</span> <span class="fu">as.numeric</span>(P4_v2), </span>
<span id="cb2-41"><a href="#cb2-41" aria-hidden="true" tabindex="-1"></a>         <span class="at">E2_v3 =</span> <span class="fu">as.numeric</span>(E2_v3), </span>
<span id="cb2-42"><a href="#cb2-42" aria-hidden="true" tabindex="-1"></a>         <span class="at">P4_v3 =</span> <span class="fu">as.numeric</span>(P4_v3), </span>
<span id="cb2-43"><a href="#cb2-43" aria-hidden="true" tabindex="-1"></a>         <span class="at">below_LLD_E2_v2 =</span> <span class="fu">factor</span>(below_LLD_E2_v2, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"no"</span>, <span class="st">"yes"</span>, <span class="cn">NA</span>)), </span>
<span id="cb2-44"><a href="#cb2-44" aria-hidden="true" tabindex="-1"></a>         <span class="at">below_LLD_P4_v2 =</span> <span class="fu">factor</span>(below_LLD_P4_v2, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"no"</span>, <span class="st">"yes"</span>, <span class="cn">NA</span>)), </span>
<span id="cb2-45"><a href="#cb2-45" aria-hidden="true" tabindex="-1"></a>         <span class="at">below_LLD_E2_v3 =</span> <span class="fu">factor</span>(below_LLD_E2_v3, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"no"</span>, <span class="st">"yes"</span>, <span class="cn">NA</span>)), </span>
<span id="cb2-46"><a href="#cb2-46" aria-hidden="true" tabindex="-1"></a>         <span class="at">below_LLD_P4_v3 =</span> <span class="fu">factor</span>(below_LLD_P4_v3, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"no"</span>, <span class="st">"yes"</span>, <span class="cn">NA</span>)), </span>
<span id="cb2-47"><a href="#cb2-47" aria-hidden="true" tabindex="-1"></a>         ) <span class="sc">%&gt;%</span></span>
<span id="cb2-48"><a href="#cb2-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">E2_v2 =</span> <span class="fu">ifelse</span>(below_LLD_E2_v2 <span class="sc">==</span> <span class="st">"yes"</span>, <span class="dv">10</span>, E2_v2), </span>
<span id="cb2-49"><a href="#cb2-49" aria-hidden="true" tabindex="-1"></a>         <span class="at">E2_v3 =</span> <span class="fu">ifelse</span>(below_LLD_E2_v3 <span class="sc">==</span> <span class="st">"yes"</span>, <span class="dv">10</span>, E2_v3), </span>
<span id="cb2-50"><a href="#cb2-50" aria-hidden="true" tabindex="-1"></a>         <span class="at">P4_v2 =</span> <span class="fu">ifelse</span>(below_LLD_P4_v2 <span class="sc">==</span> <span class="st">"yes"</span>, <span class="fl">0.05</span>, P4_v2), </span>
<span id="cb2-51"><a href="#cb2-51" aria-hidden="true" tabindex="-1"></a>         <span class="at">P4_v3 =</span> <span class="fu">ifelse</span>(below_LLD_P4_v2 <span class="sc">==</span> <span class="st">"yes"</span>, <span class="fl">0.05</span>, P4_v3))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-hash="prepare-data-sets_cache/html/tbl-BV-samples_ed6cd6bcf37166658f74053f7866a4b1">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>df_BV <span class="ot">&lt;-</span> df_clin <span class="sc">%&gt;%</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(PatID, BV_v2, BV_v3) <span class="sc">%&gt;%</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">!</span>PatID, <span class="at">names_to =</span> <span class="st">"V"</span>, <span class="at">values_to =</span> <span class="st">"BV"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">V =</span> <span class="fu">ifelse</span>(V <span class="sc">==</span> <span class="st">"BV_v2"</span>, <span class="st">"V2"</span>, <span class="st">"V3"</span>))</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>df_BV <span class="sc">%&gt;%</span></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">!</span>PatID) <span class="sc">%&gt;%</span></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tbl_summary</span>(<span class="at">by =</span> V, </span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>              <span class="at">missing_text =</span> <span class="st">"(Missing)"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_kable_extra</span>(<span class="at">booktabs =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-BV-samples" class="anchored">
<table class="table table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;1: Number of samples by BV characteristics in V2 and V3 (all samples available in meta data).</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">Characteristic</th>
<th style="text-align: center;" data-quarto-table-cell-role="th">V2, N = 121</th>
<th style="text-align: center;" data-quarto-table-cell-role="th">V3, N = 121</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">BV</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">0</td>
<td style="text-align: center;">66 (56%)</td>
<td style="text-align: center;">52 (46%)</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">1</td>
<td style="text-align: center;">28 (24%)</td>
<td style="text-align: center;">26 (23%)</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">2</td>
<td style="text-align: center;">24 (20%)</td>
<td style="text-align: center;">34 (30%)</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(Missing)</td>
<td style="text-align: center;">3</td>
<td style="text-align: center;">9</td>
</tr>
</tbody><tfoot>
<tr class="even">
<td style="text-align: left; padding: 0;"><sup>1</sup> n (%)</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
</tr>
</tfoot>

</table>
</div>


</div>
</div>
<div class="cell" data-hash="prepare-data-sets_cache/html/tbl-BV-samples-na-rm_2cb13cade2c43c22cbd0f24897c7c68f">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>df_BV <span class="ot">&lt;-</span> df_clin <span class="sc">%&gt;%</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(PatID, BV_v2, BV_v3) <span class="sc">%&gt;%</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">na.omit</span>() <span class="sc">%&gt;%</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">!</span>PatID, <span class="at">names_to =</span> <span class="st">"V"</span>, <span class="at">values_to =</span> <span class="st">"BV"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">V =</span> <span class="fu">ifelse</span>(V <span class="sc">==</span> <span class="st">"BV_v2"</span>, <span class="st">"V2"</span>, <span class="st">"V3"</span>))</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>df_BV <span class="sc">%&gt;%</span></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">!</span>PatID) <span class="sc">%&gt;%</span></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tbl_summary</span>(<span class="at">by =</span> V, </span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>              <span class="at">missing_text =</span> <span class="st">"(Missing)"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_kable_extra</span>(<span class="at">booktabs =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-BV-samples-na-rm" class="anchored">
<table class="table table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;2: Number of samples by BV characteristics in V2 and V3 (samples with missing BV entries removed).</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">Characteristic</th>
<th style="text-align: center;" data-quarto-table-cell-role="th">V2, N = 109</th>
<th style="text-align: center;" data-quarto-table-cell-role="th">V3, N = 109</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">BV</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">0</td>
<td style="text-align: center;">60 (55%)</td>
<td style="text-align: center;">52 (48%)</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">1</td>
<td style="text-align: center;">27 (25%)</td>
<td style="text-align: center;">25 (23%)</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">2</td>
<td style="text-align: center;">22 (20%)</td>
<td style="text-align: center;">32 (29%)</td>
</tr>
</tbody><tfoot>
<tr class="odd">
<td style="text-align: left; padding: 0;"><sup>1</sup> n (%)</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
</tr>
</tfoot>

</table>
</div>


</div>
</div>
</section>
<section id="count-data" class="level1">
<h1>Count data</h1>
<div class="cell" data-hash="prepare-data-sets_cache/html/unnamed-chunk-5_c2302fccaf2fe464c1d4d955129ed6aa">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># load count data</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>file_counts_v2 <span class="ot">&lt;-</span> <span class="st">"../data/raw/BEA22P129_KB/counts/subreadCounts_hg38ens_minus_frag.txt"</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>file_counts_v3 <span class="ot">&lt;-</span> <span class="st">"../data/raw/BEA22P129_KB/counts_BEA19P035_KB/subreadCounts_hg38ens_minus.txt"</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>df_counts_v2 <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(file_counts_v2, <span class="at">comment =</span> <span class="st">"#"</span>)</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>df_counts_v3 <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(file_counts_v3, <span class="at">comment =</span> <span class="st">"#"</span>)</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="co"># check dimension</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a><span class="co"># print(dim(df_counts_v2))</span></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a><span class="co"># print(dim(df_counts_v3))</span></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a><span class="co"># check if order of rows is the same (yes, it is)</span></span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a><span class="co"># summary(rownames(df_counts_v2) == rownames(df_counts_v3))</span></span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a><span class="co"># load key data to map BEA samples to PatID</span></span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a>file <span class="ot">&lt;-</span> <span class="st">"../data/box/KEY_V2_BEA22P129_KB_samples_.xlsx"</span></span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a>df_keys_v2 <span class="ot">&lt;-</span> <span class="fu">read_xlsx</span>(file)</span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a>df_keys_v2 <span class="ot">&lt;-</span> df_keys_v2 <span class="sc">%&gt;%</span></span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(File, PatID) <span class="sc">%&gt;%</span></span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">countfile_v2 =</span> File) <span class="sc">%&gt;%</span></span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">PatID =</span> <span class="fu">as.character</span>(PatID))</span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a>file <span class="ot">&lt;-</span> <span class="st">"../data/box/KEY_V3_BEA19P035_KB_samples-modified-filled-in-missing-PatID3943.xlsx"</span></span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true" tabindex="-1"></a>df_keys_v3 <span class="ot">&lt;-</span> <span class="fu">read_xlsx</span>(file)</span>
<span id="cb5-25"><a href="#cb5-25" aria-hidden="true" tabindex="-1"></a>df_keys_v3 <span class="ot">&lt;-</span> df_keys_v3 <span class="sc">%&gt;%</span></span>
<span id="cb5-26"><a href="#cb5-26" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(File, PatID) <span class="sc">%&gt;%</span></span>
<span id="cb5-27"><a href="#cb5-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">countfile_v3 =</span> File) <span class="sc">%&gt;%</span></span>
<span id="cb5-28"><a href="#cb5-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">PatID =</span> <span class="fu">as.character</span>(PatID))</span>
<span id="cb5-29"><a href="#cb5-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-30"><a href="#cb5-30" aria-hidden="true" tabindex="-1"></a><span class="co"># add key data to df_clin</span></span>
<span id="cb5-31"><a href="#cb5-31" aria-hidden="true" tabindex="-1"></a>df_meta <span class="ot">&lt;-</span> df_clin <span class="sc">%&gt;%</span></span>
<span id="cb5-32"><a href="#cb5-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(df_keys_v2) <span class="sc">%&gt;%</span></span>
<span id="cb5-33"><a href="#cb5-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(df_keys_v3) <span class="sc">%&gt;%</span></span>
<span id="cb5-34"><a href="#cb5-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">countflag_v2v3 =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(countfile_v2) <span class="sc">|</span> <span class="fu">is.na</span>(countfile_v3), <span class="dv">0</span>, <span class="dv">1</span>)) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-hash="prepare-data-sets_cache/html/tbl-BV-samples-na-rm-v2_4e5e24b82553b05a8675c7035a5fff6c">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>df_BV <span class="ot">&lt;-</span> df_meta <span class="sc">%&gt;%</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(PatID, BV_v2, countfile_v2) <span class="sc">%&gt;%</span> </span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">na.omit</span>() <span class="sc">%&gt;%</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">!</span>countfile_v2) <span class="sc">%&gt;%</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">!</span><span class="fu">c</span>(PatID), <span class="at">names_to =</span> <span class="st">"V"</span>, <span class="at">values_to =</span> <span class="st">"BV"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">V =</span> <span class="fu">ifelse</span>(V <span class="sc">==</span> <span class="st">"BV_v2"</span>, <span class="st">"V2"</span>, <span class="st">"V3"</span>)) </span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>df_BV <span class="sc">%&gt;%</span></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">!</span>PatID) <span class="sc">%&gt;%</span></span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tbl_summary</span>(<span class="at">by =</span> V, </span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a>              <span class="at">missing_text =</span> <span class="st">"(Missing)"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_kable_extra</span>(<span class="at">booktabs =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-BV-samples-na-rm-v2" class="anchored">
<table class="table table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;3: Number of samples by BV characteristics in V2, for which both BV and RNA-seq count data are available.</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">Characteristic</th>
<th style="text-align: center;" data-quarto-table-cell-role="th">V2, N = 110</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">BV</td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">0</td>
<td style="text-align: center;">63 (57%)</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">1</td>
<td style="text-align: center;">27 (25%)</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">2</td>
<td style="text-align: center;">20 (18%)</td>
</tr>
</tbody><tfoot>
<tr class="odd">
<td style="text-align: left; padding: 0;"><sup>1</sup> n (%)</td>
<td style="text-align: center;"></td>
</tr>
</tfoot>

</table>
</div>


</div>
</div>
<div class="cell" data-hash="prepare-data-sets_cache/html/tbl-BV-samples-na-rm-v3_6aec6943c2dd22540d236928e6474502">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>df_BV <span class="ot">&lt;-</span> df_meta <span class="sc">%&gt;%</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(PatID, BV_v3, countfile_v3) <span class="sc">%&gt;%</span> </span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">na.omit</span>() <span class="sc">%&gt;%</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">!</span>countfile_v3) <span class="sc">%&gt;%</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">!</span><span class="fu">c</span>(PatID), <span class="at">names_to =</span> <span class="st">"V"</span>, <span class="at">values_to =</span> <span class="st">"BV"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">V =</span> <span class="fu">ifelse</span>(V <span class="sc">==</span> <span class="st">"BV_v2"</span>, <span class="st">"V2"</span>, <span class="st">"V3"</span>)) </span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>df_BV <span class="sc">%&gt;%</span></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">!</span>PatID) <span class="sc">%&gt;%</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tbl_summary</span>(<span class="at">by =</span> V, </span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>              <span class="at">missing_text =</span> <span class="st">"(Missing)"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_kable_extra</span>(<span class="at">booktabs =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-BV-samples-na-rm-v3" class="anchored">
<table class="table table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;4: Number of samples by BV characteristics in V3, for which both BV and RNA-seq count data are available.</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">Characteristic</th>
<th style="text-align: center;" data-quarto-table-cell-role="th">V3, N = 96</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">BV</td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">0</td>
<td style="text-align: center;">43 (45%)</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">1</td>
<td style="text-align: center;">23 (24%)</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">2</td>
<td style="text-align: center;">30 (31%)</td>
</tr>
</tbody><tfoot>
<tr class="odd">
<td style="text-align: left; padding: 0;"><sup>1</sup> n (%)</td>
<td style="text-align: center;"></td>
</tr>
</tfoot>

</table>
</div>


</div>
</div>
<div class="cell" data-hash="prepare-data-sets_cache/html/tbl-BV-samples-na-rm-v2-v3_9a90a872c5a06058d73f9c3f02ac7f9f">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>df_BV <span class="ot">&lt;-</span> df_meta <span class="sc">%&gt;%</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(PatID, BV_v2, BV_v3, countfile_v2, countfile_v3) <span class="sc">%&gt;%</span> </span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">na.omit</span>() <span class="sc">%&gt;%</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">!</span><span class="fu">c</span>(countfile_v2, countfile_v3)) <span class="sc">%&gt;%</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">!</span><span class="fu">c</span>(PatID), <span class="at">names_to =</span> <span class="st">"V"</span>, <span class="at">values_to =</span> <span class="st">"BV"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">V =</span> <span class="fu">ifelse</span>(V <span class="sc">==</span> <span class="st">"BV_v2"</span>, <span class="st">"V2"</span>, <span class="st">"V3"</span>)) </span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>df_BV <span class="sc">%&gt;%</span></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">!</span>PatID) <span class="sc">%&gt;%</span></span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tbl_summary</span>(<span class="at">by =</span> V, </span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>              <span class="at">missing_text =</span> <span class="st">"(Missing)"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_kable_extra</span>(<span class="at">booktabs =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-BV-samples-na-rm-v2-v3" class="anchored">
<table class="table table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;5: Number of samples by BV characteristics in V2 and V3, for which both BV and RNA-seq count data are available at both V2 and V3.</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">Characteristic</th>
<th style="text-align: center;" data-quarto-table-cell-role="th">V2, N = 88</th>
<th style="text-align: center;" data-quarto-table-cell-role="th">V3, N = 88</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">BV</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">0</td>
<td style="text-align: center;">48 (55%)</td>
<td style="text-align: center;">42 (48%)</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">1</td>
<td style="text-align: center;">22 (25%)</td>
<td style="text-align: center;">20 (23%)</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">2</td>
<td style="text-align: center;">18 (20%)</td>
<td style="text-align: center;">26 (30%)</td>
</tr>
</tbody><tfoot>
<tr class="odd">
<td style="text-align: left; padding: 0;"><sup>1</sup> n (%)</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
</tr>
</tfoot>

</table>
</div>


</div>
</div>
</section>
<section id="subset-count-data" class="level1">
<h1>Subset count data</h1>
<div class="cell" data-hash="prepare-data-sets_cache/html/unnamed-chunk-9_0ecc3be28ff71ff8eed6bf826c1b7a81">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># select matching count data to clinical data</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>df_meta_subset <span class="ot">&lt;-</span> df_meta <span class="sc">%&gt;%</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(PatID, PatIDPx, BV_v2, BV_v3, countfile_v2, countfile_v3) <span class="sc">%&gt;%</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">na.omit</span>()</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>cols_geneinfo <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Geneid"</span>,  <span class="st">"Chr"</span>, <span class="st">"Start"</span>, <span class="st">"End"</span>, <span class="st">"Strand"</span>, <span class="st">"Length"</span>)</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>cols_v2 <span class="ot">&lt;-</span> <span class="fu">c</span>(cols_geneinfo, df_meta_subset<span class="sc">$</span>countfile_v2)</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>cols_v3 <span class="ot">&lt;-</span> <span class="fu">c</span>(cols_geneinfo, df_meta_subset<span class="sc">$</span>countfile_v3)</span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>df_counts_matched_v2 <span class="ot">&lt;-</span> df_counts_v2 <span class="sc">%&gt;%</span></span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(cols_v2))</span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a>df_counts_matched_v3 <span class="ot">&lt;-</span> df_counts_v3 <span class="sc">%&gt;%</span></span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(cols_v3))</span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a><span class="co"># check dimensions</span></span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a><span class="co"># dim(df_counts_matched_v2)</span></span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a><span class="co"># dim(df_counts_matched_v3)</span></span>
<span id="cb9-19"><a href="#cb9-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-20"><a href="#cb9-20" aria-hidden="true" tabindex="-1"></a><span class="co"># # check if order of the columns is matching meta file</span></span>
<span id="cb9-21"><a href="#cb9-21" aria-hidden="true" tabindex="-1"></a><span class="co"># colnames(df_counts_matched_v2) == cols_v2</span></span>
<span id="cb9-22"><a href="#cb9-22" aria-hidden="true" tabindex="-1"></a><span class="co"># colnames(df_counts_matched_v3) == cols_v3</span></span>
<span id="cb9-23"><a href="#cb9-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-24"><a href="#cb9-24" aria-hidden="true" tabindex="-1"></a><span class="co"># change count column names to PatID (V2)</span></span>
<span id="cb9-25"><a href="#cb9-25" aria-hidden="true" tabindex="-1"></a>df_columns_v2 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">countfile_v2 =</span> cols_v2)</span>
<span id="cb9-26"><a href="#cb9-26" aria-hidden="true" tabindex="-1"></a>df_columns_v2 <span class="ot">&lt;-</span> df_columns_v2 <span class="sc">%&gt;%</span></span>
<span id="cb9-27"><a href="#cb9-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(df_meta_subset[, <span class="fu">c</span>(<span class="st">"PatID"</span>, <span class="st">"countfile_v2"</span>)]) <span class="sc">%&gt;%</span> </span>
<span id="cb9-28"><a href="#cb9-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">columns_new =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(PatID), countfile_v2, PatID))</span>
<span id="cb9-29"><a href="#cb9-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-30"><a href="#cb9-30" aria-hidden="true" tabindex="-1"></a>df_counts_matched_v2_renamed <span class="ot">&lt;-</span> df_counts_matched_v2</span>
<span id="cb9-31"><a href="#cb9-31" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(df_counts_matched_v2_renamed) <span class="ot">&lt;-</span> df_columns_v2<span class="sc">$</span>columns_new</span>
<span id="cb9-32"><a href="#cb9-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-33"><a href="#cb9-33" aria-hidden="true" tabindex="-1"></a><span class="co"># change count column names to PatID (V3)</span></span>
<span id="cb9-34"><a href="#cb9-34" aria-hidden="true" tabindex="-1"></a>df_columns_v3 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">countfile_v3 =</span> cols_v3)</span>
<span id="cb9-35"><a href="#cb9-35" aria-hidden="true" tabindex="-1"></a>df_columns_v3 <span class="ot">&lt;-</span> df_columns_v3 <span class="sc">%&gt;%</span></span>
<span id="cb9-36"><a href="#cb9-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(df_meta_subset[, <span class="fu">c</span>(<span class="st">"PatID"</span>, <span class="st">"countfile_v3"</span>)]) <span class="sc">%&gt;%</span> </span>
<span id="cb9-37"><a href="#cb9-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">columns_new =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(PatID), countfile_v3, PatID))</span>
<span id="cb9-38"><a href="#cb9-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-39"><a href="#cb9-39" aria-hidden="true" tabindex="-1"></a>df_counts_matched_v3_renamed <span class="ot">&lt;-</span> df_counts_matched_v3</span>
<span id="cb9-40"><a href="#cb9-40" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(df_counts_matched_v3_renamed) <span class="ot">&lt;-</span> df_columns_v3<span class="sc">$</span>columns_new</span>
<span id="cb9-41"><a href="#cb9-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-42"><a href="#cb9-42" aria-hidden="true" tabindex="-1"></a><span class="co"># combine V2 and V3 count data</span></span>
<span id="cb9-43"><a href="#cb9-43" aria-hidden="true" tabindex="-1"></a>df_counts_v2_v3 <span class="ot">&lt;-</span> df_counts_matched_v2_renamed <span class="sc">%&gt;%</span></span>
<span id="cb9-44"><a href="#cb9-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(df_counts_matched_v3_renamed, <span class="at">by =</span> cols_geneinfo, <span class="at">suffix =</span> <span class="fu">c</span>(<span class="st">"_V2"</span>, <span class="st">"_V3"</span>))</span>
<span id="cb9-45"><a href="#cb9-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-46"><a href="#cb9-46" aria-hidden="true" tabindex="-1"></a><span class="co"># preview count data</span></span>
<span id="cb9-47"><a href="#cb9-47" aria-hidden="true" tabindex="-1"></a>df_counts_matched_v2_renamed <span class="sc">%&gt;%</span></span>
<span id="cb9-48"><a href="#cb9-48" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(<span class="st">"Chr"</span>, <span class="st">"Start"</span>, <span class="st">"End"</span>, <span class="st">"Strand"</span>, <span class="st">"Length"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb9-49"><a href="#cb9-49" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">11</span>) <span class="sc">%&gt;%</span></span>
<span id="cb9-50"><a href="#cb9-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">head</span>() <span class="sc">%&gt;%</span></span>
<span id="cb9-51"><a href="#cb9-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">datatable</span>(<span class="at">caption =</span> <span class="fu">paste</span>(<span class="st">"Preview of the count data corresponding to V2 samples: "</span>, <span class="fu">nrow</span>(df_counts_v2), <span class="st">" genes"</span>, <span class="at">sep=</span><span class="st">""</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">

<div class="datatables html-widget html-fill-item" id="htmlwidget-3f2f495a2a889dbd0b2b" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-3f2f495a2a889dbd0b2b">{"x":{"filter":"none","vertical":false,"caption":"<caption>Preview of the count data corresponding to V2 samples: 60671 genes<\/caption>","data":[["1","2","3","4","5","6"],["ENSG00000223972","ENSG00000227232","ENSG00000278267","ENSG00000243485","ENSG00000284332","ENSG00000237613"],[0,1,0,0,0,0],[0,2,0,0,0,0],[0,2,0,0,0,0],[0,5,1,0,0,0],[0,1,0,0,0,0],[0,4,2,0,0,0],[0,6,0,0,0,0],[0,0,0,0,0,0],[0,1,0,0,0,0],[0,2,0,0,0,0]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Geneid<\/th>\n      <th>1907<\/th>\n      <th>1954<\/th>\n      <th>1996<\/th>\n      <th>2051<\/th>\n      <th>2140<\/th>\n      <th>2296<\/th>\n      <th>2586<\/th>\n      <th>2778<\/th>\n      <th>2915<\/th>\n      <th>2928<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6,7,8,9,10,11]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>df_counts_matched_v3_renamed <span class="sc">%&gt;%</span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(<span class="st">"Chr"</span>, <span class="st">"Start"</span>, <span class="st">"End"</span>, <span class="st">"Strand"</span>, <span class="st">"Length"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">11</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">head</span>() <span class="sc">%&gt;%</span></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">datatable</span>(<span class="at">caption =</span> <span class="fu">paste</span>(<span class="st">"Preview of the count data corresponding to V3 samples: "</span>, <span class="fu">nrow</span>(df_counts_v3), <span class="st">" genes"</span>, <span class="at">sep=</span><span class="st">""</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">

<div class="datatables html-widget html-fill-item" id="htmlwidget-ab9e912b3367ca87b402" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-ab9e912b3367ca87b402">{"x":{"filter":"none","vertical":false,"caption":"<caption>Preview of the count data corresponding to V3 samples: 60671 genes<\/caption>","data":[["1","2","3","4","5","6"],["ENSG00000223972","ENSG00000227232","ENSG00000278267","ENSG00000243485","ENSG00000284332","ENSG00000237613"],[0,2,0,0,0,0],[0,0,0,0,0,0],[0,4,0,0,0,0],[0,2,0,0,0,0],[0,0,0,0,0,0],[0,4,2,0,0,0],[0,0,0,0,0,0],[0,1,1,0,0,0],[0,0,0,0,0,0],[0,5,1,0,0,0]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Geneid<\/th>\n      <th>1907<\/th>\n      <th>1954<\/th>\n      <th>1996<\/th>\n      <th>2051<\/th>\n      <th>2140<\/th>\n      <th>2296<\/th>\n      <th>2586<\/th>\n      <th>2778<\/th>\n      <th>2915<\/th>\n      <th>2928<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6,7,8,9,10,11]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>df_counts_v2_v3 <span class="sc">%&gt;%</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(<span class="st">"Chr"</span>, <span class="st">"Start"</span>, <span class="st">"End"</span>, <span class="st">"Strand"</span>, <span class="st">"Length"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">11</span>, <span class="dv">100</span> <span class="sc">:</span> <span class="dv">110</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">head</span>() <span class="sc">%&gt;%</span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">datatable</span>(<span class="at">caption =</span> <span class="fu">paste</span>(<span class="st">"Preview of the combined V2 and V3 count data: "</span>, <span class="fu">nrow</span>(df_counts_v3), <span class="st">" genes"</span>, <span class="at">sep=</span><span class="st">""</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">

<div class="datatables html-widget html-fill-item" id="htmlwidget-7409835944359a7a2a43" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-7409835944359a7a2a43">{"x":{"filter":"none","vertical":false,"caption":"<caption>Preview of the combined V2 and V3 count data: 60671 genes<\/caption>","data":[["1","2","3","4","5","6"],["ENSG00000223972","ENSG00000227232","ENSG00000278267","ENSG00000243485","ENSG00000284332","ENSG00000237613"],[0,1,0,0,0,0],[0,2,0,0,0,0],[0,2,0,0,0,0],[0,5,1,0,0,0],[0,1,0,0,0,0],[0,4,2,0,0,0],[0,6,0,0,0,0],[0,0,0,0,0,0],[0,1,0,0,0,0],[0,2,0,0,0,0],[0,2,0,0,0,0],[0,0,0,0,0,0],[0,3,0,0,0,0],[0,4,0,0,0,0],[0,3,0,0,0,0],[0,3,0,0,0,0],[0,10,3,0,0,0],[0,1,0,0,0,0],[0,4,2,0,0,0],[0,6,3,0,0,0],[0,2,1,0,0,0]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Geneid<\/th>\n      <th>1907_V2<\/th>\n      <th>1954_V2<\/th>\n      <th>1996_V2<\/th>\n      <th>2051_V2<\/th>\n      <th>2140_V2<\/th>\n      <th>2296_V2<\/th>\n      <th>2586_V2<\/th>\n      <th>2778_V2<\/th>\n      <th>2915_V2<\/th>\n      <th>2928_V2<\/th>\n      <th>2984_V3<\/th>\n      <th>3005_V3<\/th>\n      <th>3050_V3<\/th>\n      <th>3058_V3<\/th>\n      <th>3101_V3<\/th>\n      <th>3190_V3<\/th>\n      <th>3327_V3<\/th>\n      <th>3349_V3<\/th>\n      <th>3423_V3<\/th>\n      <th>3538_V3<\/th>\n      <th>3751_V3<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</section>
<section id="annotation-data" class="level1">
<h1>Annotation data</h1>
<div class="cell" data-hash="prepare-data-sets_cache/html/unnamed-chunk-10_56ccd7313151eb565867ee5b9649957f">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Use Ensembl</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>ensembl <span class="ot">=</span> <span class="fu">useMart</span>(<span class="st">"ensembl"</span>)</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="co"># listDatasets(ensembl)</span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a><span class="co"># as the main Ensembl portal is currently unavailable, one needs to specify alternative portal e.g. jul2015.archive.ensembl.org</span></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>ensemble_mart<span class="ot">=</span><span class="fu">useMart</span>(<span class="at">biomart=</span><span class="st">"ENSEMBL_MART_ENSEMBL"</span>, <span class="at">dataset=</span><span class="st">"hsapiens_gene_ensembl"</span>)<span class="co">#, host = "jul2015.archive.ensembl.org")</span></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a><span class="do">## available attributes available (uncomment below)</span></span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a><span class="co"># head(listAttributes(ensemble_mart))</span></span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a><span class="co"># ensembl_attributes &lt;- listAttributes(ensemble.mart)</span></span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a><span class="do">## save attritbutes to the file</span></span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a><span class="co"># write.table(listAttributes(ensemble.mart), file='../data/processed/listAttributes.txt', sep='\t', quote=F, row.names=F, col.names=F)</span></span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a><span class="do">## Retrieving data frame for the selected attributes</span></span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a>my_attributes <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"ensembl_gene_id"</span>, <span class="st">"entrezgene_id"</span>, <span class="st">"chromosome_name"</span>, <span class="st">"start_position"</span>,  <span class="st">"end_position"</span>, <span class="st">"strand"</span>, <span class="st">"hgnc_symbol"</span>, <span class="st">"external_gene_name"</span>, <span class="st">"description"</span>, <span class="st">"gene_biotype"</span>)</span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a>df_ensembl <span class="ot">&lt;-</span> <span class="fu">getBM</span>(<span class="at">attributes=</span>my_attributes, <span class="at">mart=</span>ensemble_mart)</span>
<span id="cb12-18"><a href="#cb12-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-19"><a href="#cb12-19" aria-hidden="true" tabindex="-1"></a><span class="do">## Add information to the ensemble genes</span></span>
<span id="cb12-20"><a href="#cb12-20" aria-hidden="true" tabindex="-1"></a><span class="do">## Genes to annotate</span></span>
<span id="cb12-21"><a href="#cb12-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-22"><a href="#cb12-22" aria-hidden="true" tabindex="-1"></a>df_genes <span class="ot">&lt;-</span> <span class="fu">tibble</span>(<span class="at">ensembl_gene_id =</span> df_counts_v2_v3<span class="sc">$</span>Geneid)</span>
<span id="cb12-23"><a href="#cb12-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-24"><a href="#cb12-24" aria-hidden="true" tabindex="-1"></a>df_ensmbl_annotated <span class="ot">&lt;-</span> df_genes <span class="sc">%&gt;%</span></span>
<span id="cb12-25"><a href="#cb12-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(df_ensembl) <span class="sc">%&gt;%</span></span>
<span id="cb12-26"><a href="#cb12-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(ensembl_gene_id, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb12-27"><a href="#cb12-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">gene =</span> <span class="fu">paste</span>(hgnc_symbol, <span class="st">"@"</span>, ensembl_gene_id, <span class="at">sep=</span><span class="st">""</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="save-data" class="level1">
<h1>Save data</h1>
<div class="cell" data-hash="prepare-data-sets_cache/html/unnamed-chunk-11_ec028028e673444609de42c746d4a952">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># directory</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>dir_out <span class="ot">&lt;-</span> <span class="st">"../data/processed/"</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a><span class="co"># rename final data sets</span></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a><span class="co"># df_meta</span></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a><span class="co"># df_meta_subset</span></span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>df_counts_v2 <span class="ot">&lt;-</span> df_counts_matched_v2_renamed</span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>df_counts_v3 <span class="ot">&lt;-</span> df_counts_matched_v3_renamed</span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a><span class="co">#df_counts_v2_v3</span></span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a><span class="co"># save final data sets to separate .csv files</span></span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a>file_save <span class="ot">&lt;-</span> <span class="st">"df_meta.csv"</span></span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a><span class="fu">write_csv</span>(df_meta, <span class="at">file =</span> <span class="fu">paste</span>(dir_out, file_save, <span class="at">sep =</span><span class="st">""</span>))</span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a>file_save <span class="ot">&lt;-</span> <span class="st">"df_counts_v2.csv"</span></span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a><span class="fu">write_csv</span>(df_counts_v2, <span class="at">file =</span> <span class="fu">paste</span>(dir_out, file_save, <span class="at">sep =</span><span class="st">""</span>))</span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-19"><a href="#cb13-19" aria-hidden="true" tabindex="-1"></a>file_save <span class="ot">&lt;-</span> <span class="st">"df_counts_v3.csv"</span></span>
<span id="cb13-20"><a href="#cb13-20" aria-hidden="true" tabindex="-1"></a><span class="fu">write_csv</span>(df_counts_v3, <span class="at">file =</span> <span class="fu">paste</span>(dir_out, file_save, <span class="at">sep =</span><span class="st">""</span>))</span>
<span id="cb13-21"><a href="#cb13-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-22"><a href="#cb13-22" aria-hidden="true" tabindex="-1"></a>file_save <span class="ot">&lt;-</span> <span class="st">"df_counts_v2_v3.csv"</span></span>
<span id="cb13-23"><a href="#cb13-23" aria-hidden="true" tabindex="-1"></a><span class="fu">write_csv</span>(df_counts_v2_v3, <span class="at">file =</span> <span class="fu">paste</span>(dir_out, file_save, <span class="at">sep =</span><span class="st">""</span>))</span>
<span id="cb13-24"><a href="#cb13-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-25"><a href="#cb13-25" aria-hidden="true" tabindex="-1"></a>file_save <span class="ot">&lt;-</span> <span class="st">"df_ensembl_annotated.csv"</span></span>
<span id="cb13-26"><a href="#cb13-26" aria-hidden="true" tabindex="-1"></a><span class="fu">write_csv</span>(df_ensmbl_annotated, <span class="at">file =</span> <span class="fu">paste</span>(dir_out, file_save, <span class="at">sep=</span><span class="st">""</span>))</span>
<span id="cb13-27"><a href="#cb13-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-28"><a href="#cb13-28" aria-hidden="true" tabindex="-1"></a><span class="co"># save final data sets to .RData</span></span>
<span id="cb13-29"><a href="#cb13-29" aria-hidden="true" tabindex="-1"></a>file_save <span class="ot">&lt;-</span> <span class="st">"data.RData"</span></span>
<span id="cb13-30"><a href="#cb13-30" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(df_meta, df_meta_subset, df_counts_v2, df_counts_v3, df_counts_v2_v3, df_ensmbl_annotated, <span class="at">file =</span> <span class="fu">paste</span>(dir_out, file_save, <span class="at">sep=</span><span class="st">""</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>